### ISSUE TITLE
Technologies stack selection discussion for the the Angular refactor

### ISSUE THREAD

**Issue by Bnaya** [OpenBudget/open-budget-frontend#299]
## On which language the code will be written
* Dart
* TypeScript
* ES6
* coffeescript + requirejs
* ES5 + requirejs
* any other candidate?

## what would be the build tool
* grunt
* gulp
* webpack
* any other candidate?

## what would be the css preprocessor (if any)
* less
* sass
* sass + compass ?
* plain css
* Should we use also css post processor?

## which angular version to use
* 1.4 current stable
* 1.5 current beta
* 2.0 current alpha, without clear ETA as far as i know atm

## Things to keep in mind:
* Wished browsers support
* localization
* Internationalization
* Accessibility (?)
* SEO (?)

## Coding Standards and code style
Indentation, curly braces style, css strategy etc...
and automatic tooling to enforce them across the code base and in the code editors
 
@akariv @mushon

**Comment by Bnaya** [comment]
@yonatann

**Comment by Bnaya** [comment]
ping @yonatann @akariv before we have any conclusions

**Comment by yonatann** [comment]
I suggest to keep everything as it is now and develop angular on top of it.
too many changes at once is generally a bad idea

**Comment by akariv** [comment]
@yonatann, I also thought that adding angular gradually on top of the
current project was the way to go.

However, after reconsidering I believe that writing things from scratch
with a proper base might be a better way to go.
We intend to rewrite most of the views anyway - the search page, the item
page, the what's new dashboard. It might be simpler to start from a clean
slate - I'm not even sure that it'll take more time as we won't have to
have awkward integration with the legacy code.

What do you think?

**Comment by Bnaya** [comment]
We can pick one simple component refactor it to angular and try to integrate it to the current code, and see how it goes

**Comment by yonatann** [comment]
@akariv I agree that in the long run a full blown refactor is very much needed.
I'm for it as long as everyone understands that the consequences are that no major updates will be available to the production site for a few months.

**Comment by akariv** [comment]
@yonatann I don't necessarily agree that we have to stop everything for a few months.

I think that we can plan this process so that we move pages slowly from the current architecture to the new one, and use a url scheme to separate the two.
For example, we could write a new search page with angular, which points to the current item pages.
sounds reasonable?

**Comment by Bnaya** [comment]
Angular 2 just got to Beta
http://angularjs.blogspot.co.il/2015/12/angular-2-beta.html

**Comment by Bnaya** [comment]
I'm working on making our current codebase work via webpack
https://github.com/OpenBudget/open-budget-frontend/tree/webpack-ftw

This will give us the ability to start using modern libs and tools alongside our current codebase and will greatly simplifiy our dev & build no matter what we will throw in

**Comment by akariv** [comment]
Great!

But let's no bundle it with your current big pending MR - which I want to
merge as soon as it's ready.

**Comment by Bnaya** [comment]
its still POC, i want to test it against all what we've got.
And the PR will *probably* come after the exemptions refactor pr

**Comment by Bnaya** [comment]
@pwalsh please join us here :)
